subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -47,6 +50,10 @@ public ExecOutputHandleRunner(String displayName, InputStream inputStream, Outpu
         this.completed = completed;
     }
 
+    public void setStartupRef(BuildOperationRef startupRef) {
```

Comment:
Instead of having a single public setter, could this be two methods (better names are welcome)?

- `associateWith(BuildOperationRef)`
- `clearBuildOperationAssociation()`